python/ngen_cal/src/ngen/cal/parameter.py

```diff
@@ -1,7 +1,7 @@
 from __future__ import annotations
 
-from pydantic import BaseModel, Field
-from typing import Sequence
+from pydantic import BaseModel, Field, root_validator
+from typing import Sequence, Mapping, Optional
 
 class Parameter(BaseModel, allow_population_by_field_name = True):
```

Comment:
In looking at the tests, what are your thoughts on validating that:
- `init`, `min`, and `max` are not `nan` (on the fence about this one)
- `init` is in the bounds of `min` and `max`
- `min` >= `max`? 

These seem like sane invariants that we _should_ uphold.